### ISSUE TITLE
Godot 3.3 2D Batching causing a Native Crash

### ISSUE THREAD

**Issue by ghsoares** [godotengine/godot#48109]
**Godot version:**

v3.3.stable.mono.official

**OS/device including version:**

Windows 10 Home
OpenGL ES 3.0 Renderer: Intel(R) HD Graphics 400
OpenGL ES Batching: ON

**Issue description:**

After running a scene of my project for a while, it causes a Native Crash, outputing something like this in the terminal:
```
ERROR: _translate_batches_to_larger_FVF: FATAL: Condition "!(cv)" is true.
   At: ./drivers/gles_common/rasterizer_canvas_batcher.h:3024
=================================================================
        Native Crash Reporting
=================================================================
Got a UNKNOWN while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries
used by your application.
=================================================================

=================================================================
        Managed Stacktrace:
=================================================================
=================================================================
```
After some testing, I discovered that the combination of Line2D with a higher number of points + a shader that I wrote would cause the crash. In my tests it took 517 points to crash. The shader that I created is a grass shader with wind effect.

**Steps to reproduce:**
I'm not sure what steps is needed to reproduce this issue, but I'll link the reproduction project that I made to test with CSharp, GDScript, GLES 3.0 and GLES 2.0, all of then caused the crash.

**Minimal reproduction project:**

[BatchingCrashIssue.zip](https://github.com/godotengine/godot/files/6360930/BatchingCrashIssue.zip)

**Comment by ghsoares** [comment]
It seens that it also depends on the distance between each point

**Comment by lawnjelly** [comment]
Great, as discussed on discord earlier. Yup can confirm it seems to be to do with the batch buffer wraparound, when you increase the batch buffer size, the i gets higher before the crash. Will look in detail tomorrow. :+1:

**Comment by qarmin** [comment]
Backtrace 
```
ERROR: _translate_batches_to_larger_FVF: FATAL: Condition "!(cv)" is true.
   At: ./drivers/gles_common/rasterizer_canvas_batcher.h:2984.
handle_crash: Program crashed with signal 4
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] godots() [0x171c67a] (/home/rafal/Pulpit/godot3.2/platform/x11/crash_handler_x11.cpp:54)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7efe53327210] (??:0)
[3] void RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::_translate_batches_to_larger_FVF<RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::BatchVertexLarge, true, true, true>(unsigned int) (/home/rafal/Pulpit/godot3.2/./drivers/gles_common/rasterizer_canvas_batcher.h:2984 (discriminator 1))
[4] RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::flush_render_batches(RasterizerCanvas::Item*, RasterizerCanvas::Item*, bool&, RasterizerStorageGLES2::Material*, unsigned int) (/home/rafal/Pulpit/godot3.2/./drivers/gles_common/rasterizer_canvas_batcher.h:2442)
[5] RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::render_joined_item_commands(RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::BItemJoined const&, RasterizerCanvas::Item*, bool&, RasterizerStorageGLES2::Material*, bool) (/home/rafal/Pulpit/godot3.2/./drivers/gles_common/rasterizer_canvas_batcher.h:2548)
[6] RasterizerCanvasGLES2::render_joined_item(RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::BItemJoined const&, RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::RenderItemState&) (/home/rafal/Pulpit/godot3.2/drivers/gles2/rasterizer_canvas_gles2.cpp:2190)
[7] RasterizerCanvasGLES2::canvas_render_items_implementation(RasterizerCanvas::Item*, int, Color const&, RasterizerCanvas::Light*, Transform2D const&) (/home/rafal/Pulpit/godot3.2/drivers/gles2/rasterizer_canvas_gles2.cpp:1242)
[8] RasterizerCanvasBatcher<RasterizerCanvasGLES2, RasterizerStorageGLES2>::batch_canvas_render_items_end() (/home/rafal/Pulpit/godot3.2/./drivers/gles_common/rasterizer_canvas_batcher.h:790)
[9] RasterizerCanvasGLES2::canvas_render_items_end() (/home/rafal/Pulpit/godot3.2/drivers/gles2/rasterizer_canvas_gles2.cpp:1216)
[10] VisualServerCanvas::render_canvas(VisualServerCanvas::Canvas*, Transform2D const&, RasterizerCanvas::Light*, RasterizerCanvas::Light*, Rect2 const&, int) (/home/rafal/Pulpit/godot3.2/servers/visual/visual_server_canvas.cpp:277)
[11] VisualServerViewport::_draw_viewport(VisualServerViewport::Viewport*, ARVRInterface::Eyes) (/home/rafal/Pulpit/godot3.2/servers/visual/visual_server_viewport.cpp:245)
[12] VisualServerViewport::draw_viewports() (/home/rafal/Pulpit/godot3.2/servers/visual/visual_server_viewport.cpp:350)
[13] VisualServerRaster::draw(bool, double) (/home/rafal/Pulpit/godot3.2/servers/visual/visual_server_raster.cpp:109 (discriminator 4))
[14] VisualServerWrapMT::draw(bool, double) (/home/rafal/Pulpit/godot3.2/servers/visual/visual_server_wrap_mt.cpp:104)
[15] Main::iteration() (/home/rafal/Pulpit/godot3.2/main/main.cpp:2139)
[16] OS_X11::run() (/home/rafal/Pulpit/godot3.2/platform/x11/os_x11.cpp:3641)
[17] godots(main+0x329) [0x171306f] (/home/rafal/Pulpit/godot3.2/platform/x11/godot_x11.cpp:57)
[18] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7efe533080b3] (??:0)
[19] godots(_start+0x2e) [0x1712c8e] (??:?)
```

**Comment by akien-mga** [comment]
Fixed by #48125.
